src/notebooks/execution/cellExecutionQueue.ts

```diff
@@ -127,8 +128,14 @@ export class CellExecutionQueue implements Disposable {
 
             let executionResult: NotebookCellRunState | undefined;
             try {
-                await cellToExecute.start(kernelConnection);
-                executionResult = await cellToExecute.result;
+                if (cellToExecute.cell.notebook.isClosed) {
```

Comment:
Found while debugging the tests.
Basically we stopped some tests early (which was not right), while some cells were still getting executed.
This ensures we don't run cells after a document has been closed.
The logs showed cells with index -1 were running.